### ISSUE TITLE
Request: make app-chrome (window) transparent

### ISSUE THREAD

**Issue by jochmann** [slashlos/He3#9]
I would love for some really helpful workflows where I can put a floating browser window over other windows and see right through a web app (or png) with disabled CSS-backgrounds. Not to be confused with the opacity setting: I need the content of the browser to be fully opaque, just that where there is no background in the web page, the app window should not obstruct my view.

E.g. I could put a He3 instance of a drawing web app (with backgrounds disabled) on top of another app on my desktop, then draw on my iPad to sync with the web app on my desktop while recording my desktop screen for tutorial videos.

current default in all apps is a white background. the only native app where I can make the app background transparent is the terminal window.

I don't know if this is feasible with the current code base, but the option used to be available with objective-c apps.
https://stackoverflow.com/questions/16135485/can-the-background-for-a-mac-app-be-transparent

**Comment by slashlos** [comment]
So basically we clear the window's opaque - currently doing that, but rather than solely affect alpha (opacity), independently affect the backgroundColor to transparent. So these behaviors - translucency & clear background are independent, all other UI being the same?

**Comment by jochmann** [comment]
yes, the window background color should be transparent, so that I can see through the app window, if the content inside the browser has no background. The browser content should be fully opaque (or can be adjusted through opacity of the app window).

I don't know the UI-kit involved, but to have a clear view I think you may need to also disable a drop shadow from the standard app-UI in MacOS?

**Comment by slashlos** [comment]
The UI  - WkWebKit is a black box.  I've coded up a test but not working, and the behavior might be able to slightly tweaked.  I like this idea and work towards it, but after 1.0.4. Rather than white I was thinking the default back view image color should match its home page urls of which there are 4.  This would be in keeping with the drag border color.

So, yeah thanks for the request.

**Comment by jochmann** [comment]
Hey, thanks for considering the feature.

**Comment by slashlos** [comment]
Very interesting idea, but would have to tailored per media: web page, AV etc file, and now seemingly pdfs - who suddenly will snapshot with inverted axis: filed support with apple.

But this all bumps into the NSThemeView which is a very closed animal, and I suspect be mucking a bit and draw fire in review.  What I'll toils towards is for a web view a transparent background - probably via injection - but I'm not a web honcho, and for AV content allowing user to color the extended frame about the media.

Thanks.

**Comment by jochmann** [comment]
I'm not sure we understand each other correctly. All I'm looking for is that the browser itself (the He3-window) has a completely transparent background as an option. Much like the app window of LayerX *after a file has been dropped on to the app window*. https://www.youtube.com/watch?v=35KixjZBDjY (I would actually prefer for the top-bar to remain visible as a drag handle).

There is no tailoring involved in my proposal: A web page is rendered. If the web page itself does not have an opaque background, I can see through the transparent parts of the web page. This alone allows for quite advanced applications of browser windows in screen recordings and live streaming.

Even more functionality could be added by disabling mouse capture to allow clicking through the browser window, but that's a separate issue.

**Comment by slashlos** [comment]
Thanks for the feedback and link content to consider.
